src/Source.jl

```diff
@@ -150,6 +153,13 @@ function Source(;fullpath::Union{AbstractString,IO}="",
                               null=options.null, dateformat=Dates.ISODateTimeFormat, decimal=options.decimal,
                               datarow=options.datarow, rows=options.rows, header=options.header, types=options.types)
         end
+        if categorical
+            for i = 1:cols
+                T = columntypes[i]
+                columntypes[i] = ifelse(length(categoricals[i]) / rows_for_type_detect < .9 && T <: WeakRefString,
+                    CategoricalValue{String, UInt16}, T)
```

Comment:
I suggested using `UInt16` only if there was a mechanism to start again with a larger type in case of overflow. As long as that mechanism isn't implemented, it would be safer to use `UInt32` even if that's wasteful most of the time.

Also, incorrect indentation?

Reply:
fair enough